docker/docker.ts

```diff
@@ -224,16 +269,17 @@ async function pullCacheAsync(
     // Ensure that we're logged in to the source registry and attempt to pull each stage in turn.
     await login();
 
-    const cacheFromImages = [];
+    const cacheFromImages: string[] = [];
     const stages = (cacheFrom.stages || []).concat([""]);
     for (const stage of stages) {
         const tag = stage ? `:${stage}` : "";
         const image = `${repoUrl}${tag}`;
 
         // Try to pull the existing image if it exists.  This may fail if the image does not exist.
-        // That's fine, just move onto the next sage.
-        const { code, stdout } = await runCommandThatCanFail(
-            "docker", ["pull", image], logResource, /*reportFullCommand:*/ true);
+        // That's fine, just move onto the next stage.
+        const { code } = await runCommandThatCanFail(
+            "docker", ["pull", image], logResource,
+            /*reportFullCommand:*/ true, /*reportErrorAsWarning:*/ true);
```

Comment:
Is this fixing some latent bug in the existing code?

Reply:
I will doc.  The idea here is that 'pulling' during the cacheefrom code will fail **the command** the first time (when things aren't cached locally).  This ends up with a pulumi update that succeeds, but which prints an error at the end.  This simply changes things to not print out that **command** failure as an 'error' but instead as a 'warning'.